activerecord/lib/active_record/migration/command_recorder.rb

```diff
@@ -163,8 +163,8 @@ def invert_add_index(args)
         table, columns, options = *args
         options ||= {}
 
-        index_name = options[:name]
-        options_hash = index_name ? { name: index_name } : { column: columns }
+        options_hash = options.slice(:name, :algorithm)
+        options_hash.merge!(column: columns) if !options_hash[:name]
```

Comment:
`.merge!(column: columns)` creates extra new `column: columns` hash.
How about `options_hash[:column] = columns unless options_hash[:name]` or simply one line fix as following?

```diff
--- a/activerecord/lib/active_record/migration/command_recorder.rb
+++ b/activerecord/lib/active_record/migration/command_recorder.rb
@@ -163,6 +163,7 @@ def invert_add_index(args)
 
           index_name = options[:name]
           options_hash = index_name ? { name: index_name } : { column: columns }
+          options_hash[:algorithm] = options[:algorithm] if options[:algorithm]
 
           [:remove_index, [table, options_hash]]
         end
```

Reply:
Thanks for your comment, I agree using `merge!` was not the best way.
Updated the pr.